proposals/gc/Overview.md

```diff
@@ -169,26 +170,25 @@ function caller() {
 ```
 
 ```
-(type $func-f64-f64 (func (param $env $anyref) (param $y f64) (result f64)))
-(type $clos-f64-f64 (struct (field $code (ref $func-f64-f64)) (field $env anyref)))
-(type $inner-env (struct (field $x i32) (field $a i32)))
+(type $code-f64-f64 (func (param $env (ref $clos-f64-f64)) (param $y f64) (result f64)))
+(type $clos-f64-f64 (struct (field $code (ref $code-f64-f64)))
+(type $inner-clos (struct (extend $clos-f46-f64) (field $x i32) (field $a i32))
 
 (func $outer (param $x f64) (result (ref $clos-f64-f64))
   (ref_func $inner)
   (get_local $x)
   (f64.add (get_local $x) (f64.const 1))
-  (new $inner-env)
   (new $clos-f64-f64)
 )
 
-(func $inner (param $anyenv anyref) (param $y f64) (result f64)
-  (local $env (ref $inner-env))
-  (block $fail (result (ref anyref))
-    (set_local $env (cast_down anyref (ref $inner-env) $fail (get_local _$anyenv)))
+(func $inner (param $clos (ref $clos-f64-f64)) (param $y f64) (result f64)
+  (local $inner-clos (ref $inner-clos))
```

Comment:
`$clos` is also the formal parameter name so it's unavailable.  Still, I changed the local name to `$env`; at least it's shorter and doesn't collide with the type name.  Happy to change to something else if you think of a better name.